### ISSUE TITLE
IDE0100 triggers on `boolVariable is false` or `boolVariable is true`

### ISSUE THREAD

**Issue by KUTlime** [dotnet/roslyn#74755]
**Brief description:**

The analyzer `IDE0100` fails on syntax with `is false` or `is true`.

![image](https://github.com/user-attachments/assets/8b5b92b0-8af3-4aa4-95da-3b73998f2ac6)

**Languages applicable:**

Both.

**Code example that the analyzer should report:**

```csharp
public AndConstraint<OneOfBaseAssertions<T0, T1, T2, T3, T4, T5>> BeT0(string because = "", params object[] becauseArgs)
    {
        _ = Execute.Assertion
                   .ForCondition(Subject.IsT0 && Subject.IsT1 is false && Subject.IsT2 is false && Subject.IsT3 is false && Subject.IsT4 is false && Subject.IsT5 is false)
                   .BecauseOf(because, becauseArgs)
                   .FailWith("Expected type to be {1}{reason}, but found {0}.", Subject, typeof(T0));

        return new(this);
    }
```

**Additional information:**

The issue arose with .NET Runtime 8.0.8, or .NET SDK 8.0.4.

The build agents picks there versions ⬆️⬆️⬆️ currently.

Previously, we were picking Runtime 8.0.7 & SDK 8.0.303 where everything was fine.

**Documentation requirements:**

Probably none

**Comment by DoctorKrolic** [comment]
This behavior makes total sense for me: `a is true` -> `a`, `a is false` -> `!a` unless `a` is a nullable value type, but you provided no info about that in the description

**Comment by sharwell** [comment]
This is the [exact purpose](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0100) of IDE0100.

**Comment by KUTlime** [comment]
@DoctorKrolic I know what is the purpose of IDE010, thank you.

@sharwell What is the purpose? Leave out the choice? This is a breaking change introduced by the recent analyzer change. I don't consider `!someBoolVariable` more readable then `someBoolVariable is false`, and I do work with new/junior developers mostly.

There should be a setup for this.

**Comment by KUTlime** [comment]
@DoctorKrolic this isn't a setup that I meant, it just boolean enable/disable.

What I would like to see is some setup like

* `all`
* `ignoreIsKeyword` to trigger on use `!=` & `==`
* `none`

**Comment by DoctorKrolic** [comment]
Adding a switch every time analyzer learns some new pattern is just unreasonable. After all, `a == true` and `a is true` are _extremely_ similar, even their string representations are of the same length

**Comment by AdrianoAE** [comment]
How is this by design? Having `foo is false` is not redundant, and replacing it with `!foo` is not readable. The purpose of this analyzer was, before this breaking change, to remove redundant checks

**Comment by CyrusNajmabadi** [comment]
> and replacing it with `!foo` is not readable

!expr has been idiomatic c# since 1.0.  This is what code style analyzers do. They push people to the idiomatic patterns the team believes the community should use. 

If you disagree, there is an option: disable the analyzer.  However, it's very much by design that these continually catch cases and push users to the coding style we think is best.

**Comment by CyrusNajmabadi** [comment]
> Leave out the choice? 

Yes.  Code style is opinionated.  We have the determination of the patterns we think are best for the ecosystem.  If a user disagrees, they can disable the specific analyzer. 

> This is a breaking change introduced by the recent analyzer change. 

Style analyzers doing have breaking changes associated with them. They are info by default.  A user turning them into warnings/errors is opting into a world where they are asking for code style enforcement to continually push their code to what we consider best practices. 

> I don't consider !someBoolVariable more readable than someBoolVariable is false and I mostly work with new/junior developers, so I know that is false is more readable than !.

Then disable the analyzer. This pattern is considered more idiomatic for the team and has been that way for the language for 25 years now. :-)
